### DISCUSSION TITLE
`gasLimit` override absent from `writeContract` docs

### DISCUSSION THREAD

**Discussion by firnprotocol** [wevm/viem#719] [General]
hi, first of all, thanks for the incredible work, and congrats on the v1 release.

i noticed that the [`writeContract` docs](https://viem.sh/docs/contract/writeContract.html) have no mention of accepting a `gasLimit` manual override.

would this override be recognized if it is provided, or would it be silently ignored?

as a related point, it looks like `write` may be making an `eth_estimateGas` RPC call internally. of course, if for whatever reason we already estimated the gas, then this call is wasteful, and we'd have plenty of reason to supply it manually (naturally we should also skip the internal call to `eth_estimateGas` in this circumstance).

let me know your thoughts. thanks in advance.

**Comment by jxom** [comment]
You can pass a `gas` attribute for gas limits. Looks like we don't have that in the docs, will update!

Passing a `gas` attribute will also skip estimation.

**Reply by firnprotocol** [reply]
nice, good to hear.
